src/components/Upload/Upload.tsx

```diff
@@ -0,0 +1,97 @@
+import { Typography } from '../Typography';
+import { UploadProps } from './types';
+import color from '../../styles/colors';
+import Icon from '../Icon/Icon';
+import React, { useEffect, useRef, useState } from 'react';
+import styles from './Upload.styles';
+
+const {
+    DivStyled,
+    TextContentStyled,
+    InputStyled,
+    ImageStyled,
+    IconDivStyled,
+} = styles;
+
+const Upload: React.FC<UploadProps> = ({ theme = 'withIcon', onChange }) => {
+    const [fileSelected, setFileSelected] = useState<Blob | null>(null);
+    const inputFile = useRef<HTMLInputElement | null>(null);
+
+    useEffect(() => onChange(fileSelected), [fileSelected]);
+
+    const onClickHandler = () => {
+        inputFile.current && inputFile.current.click();
+    };
+
+    const onChangeFile = (event: React.ChangeEvent<HTMLInputElement>) => {
+        event.preventDefault();
+        const file = event.target.files && event.target.files[0];
+        setFileSelected(file);
+    };
+
+    const onDropHandler = (event: React.DragEvent<HTMLDivElement>) => {
+        event.preventDefault();
+        const file = event.dataTransfer.files[0];
+        setFileSelected(file);
+    };
+
+    return (
+        <DivStyled
+            onClick={onClickHandler}
+            onDragOver={(e) => e.preventDefault()}
+            onDrop={onDropHandler}
+            isFileSelected={Boolean(fileSelected)}
+        >
+            {fileSelected === null && (
+                <>
+                    <InputStyled
+                        type="file"
+                        id="file"
+                        ref={inputFile}
+                        onChange={onChangeFile}
+                    />
+                    {theme === 'textOnly' && (
+                        <>
+                            <Typography variant="h4" weight="400">
+                                Drag & Drop File <br /> or
+                            </Typography>
+                            <Typography
+                                variant="body16"
+                                weight="500"
+                                color={color.blue}
+                            >
+                                Browse Files
+                            </Typography>
+                        </>
+                    )}
+                    {theme === 'withIcon' && (
+                        <>
+                            <Icon svg="image" fill={color.blueSky} size={48} />
```

Comment:
i would think upload icon would better than image as other files might be added, but i now see we dont have it in the Storybook. I think Icon is a little out of sync with the design, this is good for now I will iterate later